### ISSUE TITLE
What is the pretrain scripts?

### ISSUE THREAD

**Issue by mathfinder** [mlfoundations/dclm#68]
Thank you for your excellent work. If I want to use this data for pretraining and conduct a rigorous comparison with the DCLM-BASELINE 7B model mentioned here, what hyper-parameters should I use? Could you provide the corresponding script? Thank you.

**Comment by GeorgiosSmyrnis** [comment]
Hi @mathfinder ,

You can find the training configuration files under `training/configs`. E.g. for the 7B-1x scale, the corresponding config with all the hyperpameters is https://github.com/mlfoundations/dclm/blob/main/training/configs/7b_1x_fast_2e-3_lr_5e-6_zloss.json

Please let us know if the above helps!

**Comment by GeorgiosSmyrnis** [comment]
A small follow - up: for the largest model that we trained (beyond the competition scales), we used the same hyperparameters as the 7B-2x scale, along with the cooldown process described in Appendix P of our paper.

**Comment by mathfinder** [comment]
Thanks for your reply, but I am still confused. 

The following pre-training script template

`torchrun --nproc-per-node 8 -m training.train --scale <scale> <tokenized_json> --logs <log_dir> [--remote-sync <s3_bucket>] [--chinchilla-multiplier <multiplier>] [--clean-exp] [--report-to-wandb]`

Which one should be filled in `<tokenized_json>` to align with the results reported by leaderboard?

**Comment by mathfinder** [comment]
![image](https://github.com/user-attachments/assets/d97ae481-ce44-429b-a033-a3ea39727f76)
I guess data-config is `exp_data/datasets/tokenized/c4_original.json`? Just like the following script:
`torchrun --nproc-per-node 8 -m training.train --scale="7b_2x_fast_2e-3_lr_5e-6_zloss" --data-config="exp_data/datasets/tokenized/c4_original.json"  --report-to-wandb`

**Comment by GeorgiosSmyrnis** [comment]
Hi @mathfinder ,

For the DCLM-baseline dataset, you will need to first use DCLM-baseline from here: https://data.commoncrawl.org/contrib/datacomp/DCLM-baseline/index.html and create an untokenized json file similar to the ones found in `exp_data/datasets/raw_sources`.

After doing so, you should tokenize it using the instructions in this repository. This will produce a new `json` file under `exp_data/datasets/tokenized`, which you can then use as `tokenized_json` for `--data-config`.

**Comment by mathfinder** [comment]
And by the way, could you please provide the tokenized and shuffled dataset so that we can directly reproduce the experiment?

**Comment by GeorgiosSmyrnis** [comment]
Hi @mathfinder !

- For the highlighted experiment, you don't need to do upsampling / downsampling of sources, so you don't need the `--do_sample` parameter or the associated yaml file - you can safely ignore this.
- I will check in with the rest of the team regarding the tokenized dataset - given the size of these datasets there are some considerations regarding hosting multiple versions of the data.

**Comment by Taoer1996** [comment]
> Hi @mathfinder !
> 
> * For the highlighted experiment, you don't need to do upsampling / downsampling of sources, so you don't need the `--do_sample` parameter or the associated yaml file - you can safely ignore this.
> * I will check in with the rest of the team regarding the tokenized dataset - given the size of these datasets there are some considerations regarding hosting multiple versions of the data.

It will be very helpful for reproduction with multiple version of data!

**Comment by mathfinder** [comment]
> Hi @mathfinder !
> 
> * For the highlighted experiment, you don't need to do upsampling / downsampling of sources, so you don't need the `--do_sample` parameter or the associated yaml file - you can safely ignore this.
> * I will check in with the rest of the team regarding the tokenized dataset - given the size of these datasets there are some considerations regarding hosting multiple versions of the data.

Oh, that sounds amazing! I'm really looking forward to seeing your progress.

**Comment by GeorgiosSmyrnis** [comment]
Hi @mathfinder ,

This script should work on local datasets as well, as long as you spin up a ray cluster locally. Are you encountering any specific errors when trying to do so?
